src/url-loader/fs-url-loader.ts

```diff
@@ -64,4 +66,41 @@ export class FSUrlLoader implements UrlLoader {
     }
     return this.root ? pathlib.join(this.root, pathname) : pathname;
   }
-};
+
+  async listFilesInProject(): Promise<string[]> {
+    return this._listFilesInDir('');
+  }
+
+  async _listFilesInDir(pathFromRoot: string): Promise<string[]> {
+    const files = await new Promise<string[]>((resolve, reject) => {
+      fs.readdir(
```

Comment:
I discovered `mz` and `fs-promise` the other day. `mz` is really nice. Since we use fs-extra, I think we should pull in fs-promise: https://www.npmjs.com/package/fs-promise